flask/app/groups.py

```diff
@@ -121,20 +134,20 @@ def update_group(group_code):
         # Reset policy for group in case the group did not exist
         minio_admin.set_policy(group_code, group=group_code)
 
-    try:
-        db.session.commit()
-        return jsonify(group)
-    except:
-        db.session.rollback()
-        abort(500)
+    transaction_or_abort(db.session.commit)
+    return jsonify(group)
 
 
 @groups_blueprint.route("/api/groups", methods=["POST"])
 @login_required
 @check_admin
 @validate_json
 def create_group():
+    """
+    Create a new permission group. Administrator-only.
 
+    Potential problems: MinIO operations failing result in an inconsistent state.
+    """
     group_name = request.json.get("group_name")
```

Comment:
minor - the use of this endpoint seems to be for one group at a time but if an array is passed in, the query will throw a db error

Reply:
I'm guessing the error was `'list' object has no attribute 'get'` and I think all of our endpoints are affected due to lack of detailed input validation.